src/setup/joystick.c

```diff
@@ -94,6 +95,7 @@ static txt_joystick_axis_t *y_axis_widget;
 
 static txt_window_t *calibration_window;
 static SDL_Joystick **all_joysticks = NULL;
+static int all_joysticks_len = 0;
```

Comment:
So to give some context: previously I was using SDL_NumJoysticks() but switched to this variable because the number of joysticks can change (if you plug in new joysticks, connect new ones via Bluetooth, etc.). So I added a variable to store the number of joysticks at the time all_joysticks was allocated. num_joysticks could potentially be confused with SDL_NumJoysticks(), which it isn't - it's explicitly "the length of the all_joysticks array".